### PR TITLE
Nano: Add pytorch-lightning bf16 example

### PR DESCRIPTION
**Pull request by y199387**
## Description
Add Nano BF16 Example

### 2. User API changes
N/A

### 3. Summary of the change 
N/A

### 4. How to test?
- [x] N/A

### CODE REVIEW

File: python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    # Bfloat16 Training
+    #
+    # BFloat16 is a custom 16-bit floating point format for machine learning
+    # that’s comprised of one sign bit, eight exponent bits, and seven mantissa bits.
+    # BFloat16 has a greater "dynamic range" than FP16. This means it is able to
+    # improve numerical stability than FP16 while delivering increased performance
+    # and reducing memory usage.
+    #
+    # Intel Extension for Pytorch (a.k.a IPEX) support optimizer fusion that can reduce
+    # the number of operators that will be excuted, and reduces overhead time for FP32/BF16
```
Comment (jason-dai) [inline python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py:104]: remove these two lines

File: python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    #
+    # In BigDL-Nano, you can easily enable both IPEX optimizer fusion
+    # and BFloat16 Mixed precision by setting use_ipex=True and precision='bf16'
+    trainer = Trainer(max_epochs=5, use_ipex=True, precision='bf16')
```
Comment (jason-dai) [inline python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py:108]: does bf16 has to be used with ipex?

Comment (y199387) [reply python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py:108]: No, user can just set precision to 16/32/64/'bf16' , which will enable original PyTorch-Lightning precision plugin.

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Comment by TheaperDeng** [comment]
I think we may complement https://github.com/intel-analytics/BigDL/blob/main/python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh in next PR.

File: python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py (hunk 2 above)
Comment (TheaperDeng) [reply python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py:111]: ipex + bf16 is better than plain bf16. In this example we show how to use them both (line 106-108 for plain bf16 and line 111-113 for ipex + bf16) should we split it to 2 examples?

**Comment by TheaperDeng** [comment]
will merge this PR first and fix (split, test) further comment later

**Pull request merged**